### DISCUSSION TITLE
Query Google Lighthouse redirect message: The page may not be loading as expected because your test URL was redirected

### DISCUSSION THREAD

**Discussion by daphneyyt** [GoogleChrome/lighthouse#15746] [General]
**Issue**
When running Google Lighthouse scans (Google Lighthouse v11.1.0, on Chrome DevTools) we received the following message in the Google Lighthouse report:

The page may not be loading as expected because your test URL
(https://www.test.gov.sg/ngbbizfileinternet/faces/oracle/webcenter/portalapp/pages/TransactionMain.jspx?selectedETransId=C223&selEnityType=C) was redirected to
https://www.test.gov.sg/ngbbizfileinternet/faces/oracle/webcenter/portalapp/pages/TransactionMain.jspx?selectedETransId=C223&selEnityType=C&_afrLoop=30361070260339530&_afrWindowMode=0&Adf-Window-Id=w17yls5e73m&_afrFS=16&_afrMT=screen&_afrMFW=835&_afrMFH=775&_afrMFDW=1440&_afrMFDH=900&_afrMFC=8&_afrMFCI=0&_afrMFM=0&_afrMFR=192&_afrMFG=0&_afrMFS=0&_afrMFO=0. Try testing
the second URL directly.

**Actions we had taken**
1. We tried running Google Lighthouse scans on the original URL a few times but the same message persisted.

2.  We also tried running Google Lighthouse on the 2nd URL, but the same message still appeared in the report / we were unable to test it as we were redirected to the first URL when we loaded it into the browser.

**Questions**
1. We would like to understand under what circumstances does this message occur? 

2. We would like to check on the validity of the Google Lighthouse report - would the Google Lighthouse findings still be valid despite the presence of the message?

3. What are some steps we can take to prevent the redirect from occurring?

Thank you!

**Comment by adamraine** [comment]
So I can't access the URLs you provided, but I will try to answer your questions in a general sense:

> We would like to understand under what circumstances does this message occur?

When Lighthouse starts it takes the URL in the address bar and initiates a navigation to that URL. If there was a redirect to a different URL during that process Lighthouse will report it in the warning message you see. **This does not include any ["soft" redirects](https://github.com/WICG/soft-navigations#:~:text=%E2%80%9CSoft%20navigations%E2%80%9D%20are%20JS%2Ddriven%20same%2Ddocument%20navigations%20that%20are%20using%20the%20history%20API%20or%20the%20new%20Navigation%20API%2C%20triggered%20by%20a%20user%20gesture%20and%20modifies%20the%20DOM%2C%20modifying%20the%20previous%20content%2C%20as%20well%20as%20the%20URL%20displayed%20to%20the%20user.) which is an action that changes the URL displayed in the address bar *without* performing an actual navigation**.

It's possible that the site navigation flow goes:

1. You navigate to URL 1
2. Redirect to URL 2
3. History API used to go back to URL 1 (soft navigation)
4. Lighthouse sees URL 1 in the search bar
5. Lighthouse initiates navigation to URL 1
6. Redirect to URL 2
7. ...repeat

> We would like to check on the validity of the Google Lighthouse report - would the Google Lighthouse findings still be valid despite the presence of the message?

The report is still valid, but the results are applicable to the second URL which may be unexpected in some circumstances if you are trying to get a report for the first URL.

> What are some steps we can take to prevent the redirect from occurring?

https://developer.chrome.com/docs/lighthouse/performance/redirects/#how_to_eliminate_redirects
